### PR TITLE
fix: allow multiple hook instances

### PR DESCRIPTION
**Pull request by SamuelAlev**
And add polling to the asset upload

### CODE REVIEW

File: src/react/useBlockAssets.ts [hunk 1]
```diff
@@ -50,20 +52,43 @@ const fetchAllBlockAssets = async (blockId: number) => {
 };
 
 export const useBlockAssets = (appBridge: IAppBridgeNative) => {
+    const blockId = appBridge.blockId;
```
Comment (NadiaPosch) [inline src/react/useBlockAssets.ts:55]: Is there a reason you don't get the block id via `appBridge.getBlockId()` here?

File: src/react/useBlockAssets.ts [hunk 2]
```diff
@@ -97,26 +124,47 @@ export const useBlockAssets = (appBridge: IAppBridgeNative) => {
...
         const responseJson = await response.json();
 
-        return mapDocumentBlockAssetsToBlockAssets(responseJson.data);
+        await waitForFinishedProcessing(key);
+
+        const newAssets = mapDocumentBlockAssetsToBlockAssets(responseJson.data)[key];
+        return newAssets;
+    };
+
+    const waitForFinishedProcessing = async (key: string): Promise<void> => {
```
Comment (NadiaPosch) [inline src/react/useBlockAssets.ts:133]: You could make a direct return

File: src/utilities/object.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    if (obj1Keys.length !== obj2Keys.length) {
+        return false;
+    }
+
+    for (const key of obj1Keys) {
+        if (typeof obj1[key] === 'object') {
+            if (!compareObjects(obj1[key], obj2[key])) {
+                return false;
+            }
+        } else {
```
Comment (NadiaPosch) [inline src/utilities/object.ts:15]:
```suggestion
        if (typeof obj1[key] === 'object' && !compareObjects(obj1[key], obj2[key])) {
            return false;
        } else {
```

Comment (SamuelAlev) [reply src/utilities/object.ts]: Fixed

File: src/react/useBlockAssets.ts (hunk 1 above)
Comment (SamuelAlev) [reply src/react/useBlockAssets.ts]: Fixed

File: src/react/useBlockAssets.ts (hunk 2 above)
Comment (SamuelAlev) [reply src/react/useBlockAssets.ts:133]: This is more code style preference, the whole repo is using explicit `return` so not a good idea to introduce it

File: src/react/useBlockAssets.ts [hunk 4]
```diff
@@ -50,20 +52,43 @@ const fetchAllBlockAssets = async (blockId: number) => {
...
 
+    // Fetch the block assets on mount.
+    // And add listener for block assets updates.
     useEffect(() => {
-        (async () => {
-            const blockId = appBridge.getBlockId();
-            if (blockId) {
-                const blockAssets = await fetchAllBlockAssets(blockId);
-                setBlockAssets(blockAssets);
+        const blockId = appBridge.getBlockId();
```
Comment (NadiaPosch) [inline src/react/useBlockAssets.ts:66]: This is not needed because you already have the `blockId`

**Review by NadiaPosch — APPROVED** [review]

**Pull request merged**